manim/scene/scene.py

```diff
@@ -899,8 +902,46 @@ def get_run_time(self, animations):
         else:
             return np.max([animation.run_time for animation in animations])
 
-    def play(self, *args, **kwargs):
+    def play(
+        self,
+        *args,
+        subcaption=None,
+        subcaption_duration=None,
+        subcaption_offset=0,
+        **kwargs,
+    ):
+        r"""Plays an animation in this scene.
+
+        Parameters
+        ----------
+
+        args
+            Animations to be played.
+        subcaption
+            The content of the external subcaption that should
+            be added during the animation.
+        subcaption_duration
+            The duration for which the specified subcaption is
+            added. If ``None`` (the default), the run time of the
+            animation is taken.
+        subcaption_offset
+            An offset (in seconds) for the start time of the
+            added subcaption.
+        kwargs
+            All other keywords are passed to the renderer.
+
+        """
+        start_time = self.renderer.time
         self.renderer.play(self, *args, **kwargs)
+        run_time = self.renderer.time - start_time
+        if subcaption:
+            if subcaption_duration is None:
+                subcaption_duration = run_time
+            self.add_subcaption(
+                content=subcaption,
+                duration=subcaption_duration,
+                offset=-run_time + subcaption_offset,
```

Comment:
Why are you subtracting the `run_time` here? And if this is intended, we have two different `offset`s, which is rather confusing, isn't it?

Reply:
It is intended: the subcaption is added after the animation has been played, which means that the time stamp (`self.renderer.time`) has already been updated to the time after the animation has been played. To get the subcaption to be displayed when the animation starts playing, its starting time needs to be shifted by the run time of the animation.

What do you mean with two different `offset`s?

Reply:
> What do you mean with two different `offsets`?

We have both a `subcaption_offset` and an `offset` here. I know this is quite picky but it still confused me.

Reply:
Ah – I see! I think it makes sense for the keyword argument for `Scene.play` to be called `subcaption_offset`, to explicitly make clear that we are talking about the offset of subcaptions there -- but when using `Scene.add_subcaption`, naming the corresponding parameter `subcaption_offset` is not really necessary I believe.

I could live with refactoring it, if you'd strongly prefer that. :-)